src/async_impl/client.rs

```diff
@@ -1743,7 +1757,17 @@ impl Client {
     ///
     /// This method fails whenever the supplied `Url` cannot be parsed.
     pub fn request<U: IntoUrl>(&self, method: Method, url: U) -> RequestBuilder {
-        let req = url.into_url().map(move |url| Request::new(method, url));
+        let url = if let Some(base_url) = &self.inner.base_url {
```

Comment:
what about using `Url::join`?

https://docs.rs/url/latest/url/struct.Url.html#method.join

Reply:
This is a nice idea but while I was implementing this I noticed that Url::join returns a Result. To use it I would end up needing to change the signature of `request` (unless I just unwrap it, which I definitely am not going to do), which would break the public api. It is not really worth it since this change would only relevant to someone using a base_url.

This is unfortunate since `Url` has a `base_url` [builtin](https://docs.rs/url/latest/url/struct.ParseOptions.html#method.base_url) and it would be really nice to use that. I'm open to suggestions if anyone has any ideas on how to use `Url` without breaking the public api.

Reply:
`url` here is a `crate::Result` already. It doesn't appear in the return type because it gets moved into `RequestBuilder::new` as-is.
You wouldn't need to `unwrap`, just `map`.

Reply:
> `url` here is a `crate::Result` already. It doesn't appear in the return type because it gets moved into `RequestBuilder::new` as-is. You wouldn't need to `unwrap`, just `map`.

Yeah that was an oversight on my part. RequestBuilder takes in a client and a Result. Thanks for pointing that out!